driver-core/src/main/com/mongodb/client/model/Aggregates.java

```diff
@@ -963,28 +964,37 @@ public static Bson vectorSearch(
         notNull("queryVector", queryVector);
         notNull("index", index);
         notNull("options", options);
-        return new Bson() {
-            @Override
-            public <TDocument> BsonDocument toBsonDocument(final Class<TDocument> documentClass, final CodecRegistry codecRegistry) {
-                Document specificationDoc = new Document("path", path.toValue())
-                        .append("queryVector", queryVector)
-                        .append("index", index)
-                        .append("limit", limit);
-                specificationDoc.putAll(options.toBsonDocument(documentClass, codecRegistry));
-                return new Document("$vectorSearch", specificationDoc).toBsonDocument(documentClass, codecRegistry);
-            }
+        return new VectorSearchBson<>(path, queryVector, index, limit, options);
+    }
 
-            @Override
-            public String toString() {
-                return "Stage{name=$vectorSearch"
-                        + ", path=" + path
-                        + ", queryVector=" + queryVector
-                        + ", index=" + index
-                        + ", limit=" + limit
-                        + ", options=" + options
-                        + '}';
-            }
-        };
+    /**
+     * Creates a {@code $vectorSearch} pipeline stage supported by MongoDB Atlas.
+     * You may use the {@code $meta: "vectorSearchScore"} expression, e.g., via {@link Projections#metaVectorSearchScore(String)},
+     * to extract the relevance score assigned to each found document.
+     *
+     * @param queryVector The {@linkplain Vector query vector}. The number of dimensions must match that of the {@code index}.
+     * @param path        The field to be searched.
+     * @param index       The name of the index to use.
+     * @param limit       The limit on the number of documents produced by the pipeline stage.
+     * @param options     Optional {@code $vectorSearch} pipeline stage fields.
+     * @return The {@code $vectorSearch} pipeline stage.
+     * @mongodb.atlas.manual atlas-vector-search/vector-search-stage/ $vectorSearch
+     * @mongodb.atlas.manual atlas-search/scoring/ Scoring
+     * @mongodb.server.release 6.0.11
```

Comment:
6.0.11 was [released on 2023-Oct-11](https://www.mongodb.com/docs/manual/release-notes/6.0/#6.0.11---oct-11--2023), I doubt it supports the BSON `Binary` `subtype 9`. Let's specify just 6.0, as that's what specified in `BsonBinarySubType.VECTOR`.